codecarbon/output.py

```diff
@@ -81,7 +81,13 @@ class FileOutput(BaseOutput):
     Saves experiment artifacts to a file
     """
 
-    def __init__(self, save_file_path: str):
+    def __init__(self, save_file_path: str, on_flush: str = "append"):
+        if on_flush not in {"append", "update"}:
+            raise ValueError(
+                f"Unknown `on_flush` value: {on_flush}"
+                + " (should be one of 'append' or 'update'"
+            )
+        self.on_flush: str = on_flush
```

Comment:
Isn'it better to name the parameter "on_csv_write" also for harmonization ?
And someone could want to use _update_ with flush().

Reply:
agreed for `on_csv_write`

I don't understand what you mean for the second part however

Reply:
I guess Benoît suggests that a flush action could lead to update values already stored in the csv file.
I however doubt that this would be better than append-only mode, as it could lead to desynchronization issues if the update fails.